File: Data Structures & Algorithms/top-k-elements-in-list/submission-0.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        unordered_map<int,int> map;
        for(int num:nums)
        {
            map[num]++;
        }
        priority_queue<pair<int,int>,vector<pair<int,int>>,greater<>>pq;
        for(auto &[num,count]:map)
        {
            pq.push({count,num});
            if(pq.size()>k) pq.pop();
        }
        vector <int> result;
        while(!pq.empty())
        {
            result.push_back(pq.top().second);
            pq.pop();
        }
        return result;
        
    }
};

File: Data Structures & Algorithms/majority-element/submission-0.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        freq_map={}
        
        for i in nums:
            freq_map[i]=freq_map.get(i,0)+1

        for key, value in freq_map.items():
            if value>len(nums)//2:
                return key
        